### PR TITLE
Make `get_databricks_host_creds` backward compatible and Databricks SDK profile setting fix

### PR DESCRIPTION
**Pull request by WeichenXu123**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/WeichenXu123/mlflow/pull/12234?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12234/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12234
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

This PR contains 3 modifications:

(1)
Make `get_databricks_host_creds` backward compatible:

`get_databricks_host_creds` function is widely used in Databricks product code, to avoid breaking, fix the function to keep consistent behavior with previous MLflow version, i.e. even if we can use Databricks-SDK to make authentication, the `get_databricks_host_creds` still attach token value in returned object.

(2)
A bug fix, see https://github.com/mlflow/mlflow/pull/12234#discussion_r1625735408

(3)
Added a environmental variable configuration: `MLFLOW_ENABLE_DATABRICKS_SDK`

### How is this PR tested?

- [X] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [X] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [X] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/utils/databricks_utils.py [hunk 1]
```diff
@@ -565,45 +565,31 @@ def get_databricks_host_creds(server_uri=None):
         # support various authentication ways, so that it does not provide API
         # to get credential values. Instead, we can use ``WorkspaceClient``
         # API to invoke databricks shard restful APIs.
-        ws_client = WorkspaceClient(profile=profile)
-        host = ws_client.config.host
-        return MlflowHostCreds(
-            host=host, auth_by_databricks_sdk=True, databricks_auth_profile=profile
-        )
+        WorkspaceClient(profile=profile)
```
Comment (harupy) [inline mlflow/utils/databricks_utils.py:568]: what does this line do?

File: mlflow/utils/rest_utils.py [hunk 2]
```diff
@@ -398,6 +398,12 @@ class MlflowHostCreds:
             Sets the verify param of the ``requests.request``
             function (see https://requests.readthedocs.io/en/master/api/).
             If this is set ``ignore_tls_verification`` must be false.
+        auth_by_databricks_sdk: A boolean value represent whether using Databricks SDK to
+            do authentication.
```
Comment (harupy) [inline mlflow/utils/rest_utils.py:402]: /ai fix this argument description

File: mlflow/utils/rest_utils.py [hunk 3]
```diff
@@ -398,6 +398,12 @@ class MlflowHostCreds:
...
+        databricks_auth_profile: The name of the profile used by Databricks SDK to
+            do authentication.
```
Comment (harupy) [inline mlflow/utils/rest_utils.py:404]:
```suggestion
        databricks_auth_profile: The name of the profile used by Databricks SDK for authentication
```

File: mlflow/utils/databricks_utils.py (hunk 1 above)
Comment (harupy) [reply mlflow/utils/databricks_utils.py:568]: ah I see, try-catch

File: mlflow/utils/rest_utils.py [hunk 4]
```diff
@@ -88,9 +88,12 @@ def http_request(
...
 
-        config = Config(retry_timeout_seconds=MLFLOW_DATABRICKS_ENDPOINT_HTTP_RETRY_TIMEOUT.get())
-        ws_client = WorkspaceClient(profile=host_creds.databricks_auth_profile, config=config)
-
+        config = Config(
+            profile=host_creds.databricks_auth_profile,
+            retry_timeout_seconds=MLFLOW_DATABRICKS_ENDPOINT_HTTP_RETRY_TIMEOUT.get()
+        )
+        # Note: If we use `config` param, all SDK configurations must be set in `config` object.
+        ws_client = WorkspaceClient(config=config)
```
Comment (WeichenXu123) [inline mlflow/utils/rest_utils.py:96]: Bug fix.

**Comment by WeichenXu123** [comment]
Filed new PR https://github.com/mlflow/mlflow/pull/12252 against feature branch "databricks-sdk"

**Pull request closed without merging**